src/containers/my-courses/add-course-with-input/AddCourseWithInput.tsx

```diff
@@ -11,16 +11,29 @@ import AppButton from '~/components/app-button/AppButton'
 import InputWithIcon from '~/components/input-with-icon/InputWithIcon'
 import CoursesFilterBar from '~/containers/find-course/courses-filter-bar/CoursesFilterBar'
 import { styles } from '~/containers/my-courses/add-course-with-input/AddCourseWithInput.styles'
+import CoursesFiltersDrawer from '~/containers/my-courses/courses-filters-drawer/CoursesFiltersDrawer'
+import useForm from '~/hooks/use-form'
+import { CourseFilters } from '~/types'
 
 const AddCourseWithInput = () => {
   const [inputValue, setInputValue] = useState<string>('')
+  const [isFiltersOpen, setIsFiltersOpen] = useState(false)
   const { t } = useTranslation()
 
+  const { data: filters, handleNonInputValueChange } = useForm<CourseFilters>({
+    initialValues: {
+      category: '',
+      subject: '',
+      proficiencyLevel: []
+    }
+  })
+
   const onChange = (e: ChangeEvent<HTMLInputElement>) => {
     setInputValue(e.target.value)
   }
-
   const onClear = () => setInputValue('')
+  const handleToggle = () => setIsFiltersOpen(!isFiltersOpen)
```

Comment:
I think good practice to use previous state to prevents potential issues. `setIsFiltersOpen(prevState => !prevState)`